hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/PlanCreateVisitor.java

```diff
@@ -318,6 +332,49 @@ public void onFilter(FilterPhysicalRel rel) {
         pushUpstream(filterNode);
     }
 
+    @Override
+    public void onSortMergeExchange(SortMergeExchangePhysicalRel rel) {
+        PlanNode upstreamNode = pollSingleUpstream();
+        PlanNodeSchema upstreamNodeSchema = upstreamNode.getSchema();
+
+        // Create sender and push it as a fragment.
+        int edge = nextEdge();
+
+        int id = pollId(rel);
+
+        UnicastSendPlanNode sendNode = new UnicastSendPlanNode(
+            id,
+            upstreamNode,
+            edge,
+            AllFieldsRowPartitioner.INSTANCE
+        );
+
+        addFragment(sendNode, dataMemberMapping());
+
+        List<RelFieldCollation> collations = rel.getCollation().getFieldCollations();
+        List<Expression> expressions = new ArrayList<>(collations.size());
+        List<Boolean> ascs = new ArrayList<>(collations.size());
+
+        for (RelFieldCollation collation : collations) {
+            RelFieldCollation.Direction direction = collation.getDirection();
+            int idx = collation.getFieldIndex();
+
+            expressions.add(ColumnExpression.create(idx, upstreamNodeSchema.getType(idx)));
```

Comment:
It seems that the only expression that is ever passed to `ReceiveSortMergePlanNode` is `ColumnExpression`. Do we really need to serialize them? Perhaps just `idx` would be enough (i.e. `List<Integer>` instead of `List<Expression>`)

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/18013/commits/d94b80256f44b6775758fdf16d03222daa3ea801